### DISCUSSION TITLE
Newbie question re contributing to statamic/cms

### DISCUSSION THREAD

**Discussion by jensolafkoch** [statamic/cms#4564] [Q&A] [answered]
Please be forgiving if the following is too dumb a question ... :-) Although I've been using PHP, Laravel, and Statamic for a couple of client projects in a small team I have never contributed to any bigger project yet ...

*

What I've done so far:

- forked `statamic/cms`
- cloned locally
- setup a pure Statamic site with `statamic new ...`

Then I tried to follow this approach 

[https://johnbraun.blog/posts/contributing-to-a-PHP-package](https://johnbraun.blog/posts/contributing-to-a-PHP-package)

to use my cloned local core package instead of `statamic/cms` by 

- removing `statamic/cms` from `composer.json`:

`composer remove statamic/cms`

- including a "repositories" section in the app's `composer.json` with the path to the local package folder (which differs from the package name due to my local naming structure):

```
"repositories": [
    {
      "type": "path",
      "url": "../z-pack-statamic-cms"
    }
  ]

```
- change the "name" in the local package `composer.json` to the name I used for my fork on GitHub:

`"name": "jensolafkoch/statamic-cms",`

- requiring my package in the app's `composer.json`:

`composer require jensolafkoch/statamic-cms`

*

Effect:

When I try to open the CP from the app's home page I get:

**Could not find the [statamic/cms] in your composer.lock file.** 

So clearly something went wrong ... But I cannot require both the original statamic/cms and the local renamed copy at the same time, right?

I hope someone can help? Many thanks in advance!

**Comment by jasonvarga** [comment, marked as answer]
We'll be adding a more in depth article to the docs on how to do all this.

You don't need to change the `name` in your fork's `composer.json`.
You should continue to require `statamic/cms`.

Adding the `repositories` to your app's `composer.json` to point to your local fork should be enough.
You might need to add `canonical: false`.

```json
"repositories": [
    {
        "type": "path",
        "url": "/path/to/cms",
        "canonical": false
    }
]
```

Once you've added the `repositories`, you can do `composer require "statamic/cms 3.2.x-dev"`

The `3.2.x-dev` part corresponds to the branch you have checked out.
If it's numeric, like `3.2`, you have to do `3.2.x-dev`.
If it's a word, like `mybranch` (or `feature/mybranch`) you have to do `dev-mybranch` (or `dev-feature/mybranch`)

**Reply by jensolafkoch** [reply]
Thanks a lot, Jason, does work like a charm. :-)
